src/inventory/ModelManager.cpp

```diff
@@ -372,17 +395,22 @@ void ModelManager::scanForModels(QString path) {
                 continue;
 
             QJsonObject obj = getModelInfoJsonFromDir(current);
-            if (!obj.empty()) {
-                Model model = parseModelInfo(obj);
-                if (model.path != "") {
-                    insertLocalModel(model);
-                } else {
-                    emit error(tr("Corrupted json file: %1/model_info.json. Delete or redownload.").arg(current));
-                }
-            } else {
-                // We have a folder in our models directory that doesn't contain a model. This is ok.
+            
+            // We have a folder in our models directory that doesn't contain a model. This is ok.
+            if (obj.empty())
+                continue;
+            
+            auto model = parseModelInfo(obj);
+            if (!model) {
+                emit error(tr("Corrupted json file: %1/model_info.json. Delete or redownload.").arg(current));
```

Comment:
Should we make this a messagebox where ok deletes the model? Or too much plumbing. This scenario would be quite unlikely I feel

Reply:
There is already validation on download, so this could be a corruption scenario. But more likely it is someone trying to make a model that works with translateLocally before uploading it somewhere. With that in mind, I'm adding a better description of what might be wrong with the file.